crates/libs/implement/src/gen.rs

```diff
@@ -0,0 +1,579 @@
+        if !tear_off_ptr.is_null() {
+            *interface = tear_off_ptr;
+            return ::windows_core::HRESULT(0);
+        }
+    };
+
+    parse_quote! {
+        unsafe fn QueryInterface(
+            &self,
+            iid: *const ::windows_core::GUID,
+            interface: *mut *mut ::core::ffi::c_void,
+        ) -> ::windows_core::HRESULT {
+            unsafe {
+                if iid.is_null() || interface.is_null() {
+                    return ::windows_core::imp::E_POINTER;
+                }
+
+                let iid = *iid;
+
+                let interface_ptr: *const ::core::ffi::c_void = 'found: {
+                    #identity_query
+                    #(#queries)*
+                    #dynamic_cast_query
+                    #tear_off_query
+
+                    *interface = ::core::ptr::null_mut();
+                    return ::windows_core::imp::E_NOINTERFACE;
+                };
+
+                debug_assert!(!interface_ptr.is_null());
+                *interface = interface_ptr as *mut ::core::ffi::c_void;
+                self.count.add_ref();
+                return ::windows_core::HRESULT(0);
+            }
+        }
+    }
+}
+
+/// Generates the `T::into_outer` function. This function is part of how we construct a
+/// `ComObject<T>` from a `T`.
+fn gen_into_outer(inputs: &ImplementInputs) -> syn::ImplItem {
+    let generics = &inputs.generics;
+    let impl_ident = &inputs.impl_ident;
+
+    let mut initializers = quote! {
+        identity: &#impl_ident::#generics::VTABLE_IDENTITY,
+    };
+
+    for interface_chain in inputs.interface_chains.iter() {
+        let vtbl_field_ident = &interface_chain.field_ident;
+        let vtable_const_ident = &interface_chain.vtable_const_ident;
+
+        initializers.extend(quote_spanned! {
+            interface_chain.implement.span =>
+            #vtbl_field_ident: &#impl_ident::#generics::#vtable_const_ident,
+        });
+    }
+
+    // If the type is generic then into_outer() cannot be a const fn.
+    let maybe_const = if inputs.is_generic {
+        quote!()
+    } else {
+        quote!(const)
+    };
+
+    parse_quote! {
+        // This constructs an "outer" object. This should only be used by the implementation
+        // of the outer object, never by application code.
+        //
+        // The callers of this function (`into_static` and `into_object`) are both responsible
+        // for maintaining one of our invariants: Application code never has an owned instance
+        // of the outer (implementation) type. into_static() maintains this invariant by
+        // returning a wrapped StaticComObject value, which owns its contents but never gives
+        // application code a way to mutably access its contents. This prevents the refcount
+        // shearing problem.
+        //
+        // TODO: Make it impossible for app code to call this function, by placing it in a
```

Comment:
Do we want to emit this TODO into the generated code?

Reply:
Ordinary comments are not emitted into the token stream, so this won't be emitted.  Rustdoc comments are, though.

Reply:
Feel free to keep the TODO in the code as a reminder.